From 211b73efb2a416f3b464d9d74da0aa8f93abe016 Mon Sep 17 00:00:00 2001
From: Don Olmstead <don.olmstead@sony.com>
Date: Fri, 8 Mar 2024 17:27:09 -0800
Subject: [PATCH] Non-unified build fixes, early March 2024 edition
 https://bugs.webkit.org/show_bug.cgi?id=270719

Unreviewed non-unified build fix.

* Source/JavaScriptCore/jit/JITSizeStatistics.cpp:
* Source/WebCore/Modules/model-element/ModelPlayerClient.h:
* Source/WebCore/css/CSSCrossfadeValue.h:
* Source/WebCore/css/CSSImageSetOptionValue.cpp:
* Source/WebCore/css/CSSImageSetOptionValue.h:
* Source/WebCore/css/CSSReflectValue.cpp:
* Source/WebCore/css/CSSReflectValue.h:
* Source/WebCore/css/CSSShadowValue.h:
* Source/WebCore/css/CSSValuePair.cpp:
* Source/WebCore/css/CSSValuePair.h:
* Source/WebCore/dom/SlotAssignment.h:
* Source/WebCore/dom/TrustedType.cpp:
* Source/WebCore/rendering/svg/RenderSVGPath.cpp:
* Source/WebCore/rendering/svg/RenderSVGResourceLinearGradient.cpp:
* Source/WebCore/rendering/svg/RenderSVGResourceMarker.cpp:
* Source/WebCore/rendering/svg/RenderSVGResourceMasker.cpp:
* Source/WebCore/rendering/svg/RenderSVGResourceRadialGradient.cpp:
* Source/WebKit/GPUProcess/graphics/RemoteImageBufferSet.cpp:
* Source/WebKit/Shared/WebGPU/WebGPURenderPassDepthStencilAttachment.cpp:
* Source/WebKit/UIProcess/BrowsingContextGroup.cpp:
* Source/WebKit/UIProcess/FrameProcess.cpp:
* Source/WebKit/UIProcess/ProvisionalPageProxy.cpp:

Canonical link: https://commits.webkit.org/275863@main

(cherry picked from commit 38c6ecad0ea32e80cb3e811c3ff759074006791f)
---
 Source/JavaScriptCore/jit/JITSizeStatistics.cpp               | 1 +
 Source/WebCore/Modules/model-element/ModelPlayerClient.h      | 1 +
 Source/WebCore/css/CSSCrossfadeValue.h                        | 3 ++-
 Source/WebCore/css/CSSImageSetOptionValue.cpp                 | 1 -
 Source/WebCore/css/CSSImageSetOptionValue.h                   | 4 ++--
 Source/WebCore/css/CSSReflectValue.cpp                        | 2 --
 Source/WebCore/css/CSSReflectValue.h                          | 4 ++--
 Source/WebCore/css/CSSShadowValue.h                           | 4 ++--
 Source/WebCore/css/CSSValuePair.cpp                           | 1 +
 Source/WebCore/css/CSSValuePair.h                             | 1 +
 Source/WebCore/dom/SlotAssignment.h                           | 1 +
 Source/WebCore/rendering/svg/RenderSVGPath.cpp                | 2 ++
 .../WebCore/rendering/svg/RenderSVGResourceLinearGradient.cpp | 2 ++
 Source/WebCore/rendering/svg/RenderSVGResourceMarker.cpp      | 2 ++
 Source/WebCore/rendering/svg/RenderSVGResourceMasker.cpp      | 2 ++
 .../WebCore/rendering/svg/RenderSVGResourceRadialGradient.cpp | 2 ++
 .../Shared/WebGPU/WebGPURenderPassDepthStencilAttachment.cpp  | 1 +
 Source/WebKit/UIProcess/ProvisionalPageProxy.cpp              | 1 +
 18 files changed, 25 insertions(+), 10 deletions(-)

diff --git a/Source/JavaScriptCore/jit/JITSizeStatistics.cpp b/Source/JavaScriptCore/jit/JITSizeStatistics.cpp
index f1bc2780343a..b2ffd81ee34d 100644
--- a/Source/JavaScriptCore/jit/JITSizeStatistics.cpp
+++ b/Source/JavaScriptCore/jit/JITSizeStatistics.cpp
@@ -29,6 +29,7 @@
 #if ENABLE(JIT)
 
 #include "CCallHelpers.h"
+#include "JITPlan.h"
 #include "LinkBuffer.h"
 #include <wtf/BubbleSort.h>
 #include <wtf/TZoneMallocInlines.h>
diff --git a/Source/WebCore/Modules/model-element/ModelPlayerClient.h b/Source/WebCore/Modules/model-element/ModelPlayerClient.h
index 9343d5102015..b9cf2857eee2 100644
--- a/Source/WebCore/Modules/model-element/ModelPlayerClient.h
+++ b/Source/WebCore/Modules/model-element/ModelPlayerClient.h
@@ -25,6 +25,7 @@
 
 #pragma once
 
+#include "LayerHostingContextIdentifier.h"
 #include "PlatformLayerIdentifier.h"
 #include <wtf/Forward.h>
 #include <wtf/WeakPtr.h>
diff --git a/Source/WebCore/css/CSSCrossfadeValue.h b/Source/WebCore/css/CSSCrossfadeValue.h
index eafa3bb5e828..86b87be608fe 100644
--- a/Source/WebCore/css/CSSCrossfadeValue.h
+++ b/Source/WebCore/css/CSSCrossfadeValue.h
@@ -25,11 +25,12 @@
 
 #pragma once
 
+#include "CSSPrimitiveValue.h"
 #include "CSSValue.h"
+#include <wtf/Function.h>
 
 namespace WebCore {
 
-class CSSPrimitiveValue;
 class StyleImage;
 
 namespace Style {
diff --git a/Source/WebCore/css/CSSImageSetOptionValue.cpp b/Source/WebCore/css/CSSImageSetOptionValue.cpp
index 9becc5444e10..2b7812e419a2 100644
--- a/Source/WebCore/css/CSSImageSetOptionValue.cpp
+++ b/Source/WebCore/css/CSSImageSetOptionValue.cpp
@@ -27,7 +27,6 @@
 #include "CSSImageSetOptionValue.h"
 
 #include "CSSImageValue.h"
-#include "CSSPrimitiveValue.h"
 
 namespace WebCore {
 
diff --git a/Source/WebCore/css/CSSImageSetOptionValue.h b/Source/WebCore/css/CSSImageSetOptionValue.h
index 60ef16534b24..c65ddccc04b5 100644
--- a/Source/WebCore/css/CSSImageSetOptionValue.h
+++ b/Source/WebCore/css/CSSImageSetOptionValue.h
@@ -25,13 +25,13 @@
 
 #pragma once
 
+#include "CSSPrimitiveValue.h"
 #include "CSSValue.h"
+#include <wtf/Function.h>
 #include <wtf/text/WTFString.h>
 
 namespace WebCore {
 
-class CSSPrimitiveValue;
-
 class CSSImageSetOptionValue final : public CSSValue {
 public:
     static Ref<CSSImageSetOptionValue> create(Ref<CSSValue>&&);
diff --git a/Source/WebCore/css/CSSReflectValue.cpp b/Source/WebCore/css/CSSReflectValue.cpp
index 371e6ed16087..6049d4af7d2d 100644
--- a/Source/WebCore/css/CSSReflectValue.cpp
+++ b/Source/WebCore/css/CSSReflectValue.cpp
@@ -26,8 +26,6 @@
 #include "config.h"
 #include "CSSReflectValue.h"
 
-#include "CSSPrimitiveValue.h"
-
 namespace WebCore {
 
 CSSReflectValue::CSSReflectValue(CSSValueID direction, Ref<CSSPrimitiveValue> offset, RefPtr<CSSValue> mask)
diff --git a/Source/WebCore/css/CSSReflectValue.h b/Source/WebCore/css/CSSReflectValue.h
index c67d2ad0dc4b..84f7b1d0e7e1 100644
--- a/Source/WebCore/css/CSSReflectValue.h
+++ b/Source/WebCore/css/CSSReflectValue.h
@@ -25,12 +25,12 @@
 
 #pragma once
 
+#include "CSSPrimitiveValue.h"
 #include "CSSValue.h"
+#include <wtf/Function.h>
 
 namespace WebCore {
 
-class CSSPrimitiveValue;
-
 class CSSReflectValue final : public CSSValue {
 public:
     static Ref<CSSReflectValue> create(CSSValueID direction, Ref<CSSPrimitiveValue> offset, RefPtr<CSSValue> mask);
diff --git a/Source/WebCore/css/CSSShadowValue.h b/Source/WebCore/css/CSSShadowValue.h
index 26857028cf36..7d6d25980f72 100644
--- a/Source/WebCore/css/CSSShadowValue.h
+++ b/Source/WebCore/css/CSSShadowValue.h
@@ -20,13 +20,13 @@
 
 #pragma once
 
+#include "CSSPrimitiveValue.h"
 #include "CSSValue.h"
+#include <wtf/Function.h>
 #include <wtf/RefPtr.h>
 
 namespace WebCore {
 
-class CSSPrimitiveValue;
-
 // Used for text-shadow and box-shadow
 class CSSShadowValue final : public CSSValue {
 public:
diff --git a/Source/WebCore/css/CSSValuePair.cpp b/Source/WebCore/css/CSSValuePair.cpp
index 7c64d5a9e353..a3e169a697c7 100644
--- a/Source/WebCore/css/CSSValuePair.cpp
+++ b/Source/WebCore/css/CSSValuePair.cpp
@@ -25,6 +25,7 @@
 
 #include "config.h"
 #include "CSSValuePair.h"
+
 #include <wtf/Hasher.h>
 #include <wtf/text/WTFString.h>
 
diff --git a/Source/WebCore/css/CSSValuePair.h b/Source/WebCore/css/CSSValuePair.h
index 0fc85298ca62..44c4d97c01ab 100644
--- a/Source/WebCore/css/CSSValuePair.h
+++ b/Source/WebCore/css/CSSValuePair.h
@@ -26,6 +26,7 @@
 #pragma once
 
 #include "CSSValue.h"
+#include <wtf/Function.h>
 
 namespace WebCore {
 
diff --git a/Source/WebCore/dom/SlotAssignment.h b/Source/WebCore/dom/SlotAssignment.h
index 9f9e1159e10a..478ca88bfca6 100644
--- a/Source/WebCore/dom/SlotAssignment.h
+++ b/Source/WebCore/dom/SlotAssignment.h
@@ -30,6 +30,7 @@
 #include <wtf/HashSet.h>
 #include <wtf/Vector.h>
 #include <wtf/WeakHashMap.h>
+#include <wtf/WeakHashSet.h>
 #include <wtf/WeakPtr.h>
 #include <wtf/text/AtomString.h>
 #include <wtf/text/AtomStringHash.h>
diff --git a/Source/WebCore/rendering/svg/RenderSVGPath.cpp b/Source/WebCore/rendering/svg/RenderSVGPath.cpp
index e9977db1c8b6..eba99e3a49d3 100644
--- a/Source/WebCore/rendering/svg/RenderSVGPath.cpp
+++ b/Source/WebCore/rendering/svg/RenderSVGPath.cpp
@@ -30,12 +30,14 @@
 #include "RenderSVGPath.h"
 
 #if ENABLE(LAYER_BASED_SVG_ENGINE)
+
 #include "Gradient.h"
 #include "ReferencedSVGResources.h"
 #include "RenderLayer.h"
 #include "RenderSVGResourceMarkerInlines.h"
 #include "RenderSVGShapeInlines.h"
 #include "RenderStyleInlines.h"
+#include "SVGElementTypeHelpers.h"
 #include "SVGMarkerElement.h"
 #include "SVGPathElement.h"
 #include "SVGSubpathData.h"
diff --git a/Source/WebCore/rendering/svg/RenderSVGResourceLinearGradient.cpp b/Source/WebCore/rendering/svg/RenderSVGResourceLinearGradient.cpp
index 85f79031ed4e..2b379fa4ebfe 100644
--- a/Source/WebCore/rendering/svg/RenderSVGResourceLinearGradient.cpp
+++ b/Source/WebCore/rendering/svg/RenderSVGResourceLinearGradient.cpp
@@ -23,8 +23,10 @@
 #include "RenderSVGResourceLinearGradient.h"
 
 #if ENABLE(LAYER_BASED_SVG_ENGINE)
+
 #include "RenderSVGModelObjectInlines.h"
 #include "RenderSVGResourceLinearGradientInlines.h"
+#include "SVGElementTypeHelpers.h"
 #include <wtf/IsoMallocInlines.h>
 
 namespace WebCore {
diff --git a/Source/WebCore/rendering/svg/RenderSVGResourceMarker.cpp b/Source/WebCore/rendering/svg/RenderSVGResourceMarker.cpp
index d65a81f773eb..27abd9b6e0f7 100644
--- a/Source/WebCore/rendering/svg/RenderSVGResourceMarker.cpp
+++ b/Source/WebCore/rendering/svg/RenderSVGResourceMarker.cpp
@@ -24,6 +24,7 @@
 #include "RenderSVGResourceMarker.h"
 
 #if ENABLE(LAYER_BASED_SVG_ENGINE)
+
 #include "Element.h"
 #include "ElementIterator.h"
 #include "FloatPoint.h"
@@ -34,6 +35,7 @@
 #include "RenderLayerInlines.h"
 #include "RenderSVGModelObjectInlines.h"
 #include "RenderSVGResourceMarkerInlines.h"
+#include "SVGElementTypeHelpers.h"
 #include "SVGGraphicsElement.h"
 #include "SVGLengthContext.h"
 #include "SVGRenderStyle.h"
diff --git a/Source/WebCore/rendering/svg/RenderSVGResourceMasker.cpp b/Source/WebCore/rendering/svg/RenderSVGResourceMasker.cpp
index 81321523d0e4..86c59ca6fab6 100644
--- a/Source/WebCore/rendering/svg/RenderSVGResourceMasker.cpp
+++ b/Source/WebCore/rendering/svg/RenderSVGResourceMasker.cpp
@@ -22,6 +22,7 @@
 #include "RenderSVGResourceMasker.h"
 
 #if ENABLE(LAYER_BASED_SVG_ENGINE)
+
 #include "Element.h"
 #include "ElementIterator.h"
 #include "FloatPoint.h"
@@ -33,6 +34,7 @@
 #include "RenderSVGModelObjectInlines.h"
 #include "RenderSVGResourceMaskerInlines.h"
 #include "SVGContainerLayout.h"
+#include "SVGElementTypeHelpers.h"
 #include "SVGGraphicsElement.h"
 #include "SVGLengthContext.h"
 #include "SVGRenderStyle.h"
diff --git a/Source/WebCore/rendering/svg/RenderSVGResourceRadialGradient.cpp b/Source/WebCore/rendering/svg/RenderSVGResourceRadialGradient.cpp
index 7d70fd721d73..25a9b293468e 100644
--- a/Source/WebCore/rendering/svg/RenderSVGResourceRadialGradient.cpp
+++ b/Source/WebCore/rendering/svg/RenderSVGResourceRadialGradient.cpp
@@ -23,9 +23,11 @@
 #include "RenderSVGResourceRadialGradient.h"
 
 #if ENABLE(LAYER_BASED_SVG_ENGINE)
+
 #include "RenderSVGModelObjectInlines.h"
 #include "RenderSVGResourceRadialGradientInlines.h"
 #include "RenderSVGShape.h"
+#include "SVGElementTypeHelpers.h"
 #include <wtf/IsoMallocInlines.h>
 
 namespace WebCore {
diff --git a/Source/WebKit/Shared/WebGPU/WebGPURenderPassDepthStencilAttachment.cpp b/Source/WebKit/Shared/WebGPU/WebGPURenderPassDepthStencilAttachment.cpp
index a0c521a767f1..3d4c130de7fa 100644
--- a/Source/WebKit/Shared/WebGPU/WebGPURenderPassDepthStencilAttachment.cpp
+++ b/Source/WebKit/Shared/WebGPU/WebGPURenderPassDepthStencilAttachment.cpp
@@ -31,6 +31,7 @@
 #include "WebGPUConvertFromBackingContext.h"
 #include "WebGPUConvertToBackingContext.h"
 #include <WebCore/WebGPURenderPassDepthStencilAttachment.h>
+#include <WebCore/WebGPUTextureView.h>
 
 namespace WebKit::WebGPU {
 
-- 
2.43.0

